### PR TITLE
fix(ext/node): permission prompt for missing `process.env` permissions

### LINKED ISSUE #18665: `npm:supports-color` doesn't give the right values in deno
```js
import supportsColor from "npm:supports-color";

console.log(supportsColor)

if (supportsColor.stdout) {
  console.log("Terminal stdout supports color");
}

if (supportsColor.stdout.has256) {
  console.log("Terminal stdout supports 256 colors");
}

if (supportsColor.stderr.has16m) {
  console.log("Terminal stderr supports 16 million colors (truecolor)");
}
```

### PR DESCRIPTION
**Pull request by iuioiua**
IIUC, this fixes #18665

Supercedes #20213

CC @kt3k

### CODE REVIEW

File: ext/node/polyfills/_process/process.ts [hunk 1]
```diff
@@ -36,17 +36,10 @@ export const nextTick = _nextTick;
...
-      "granted"; // for Deno Deploy
-  // Returns undefined if the env permission is unavailable
-  if (perm !== "granted") {
-    return undefined;
-  }
   try {
     return Deno.env.get(name);
   } catch (e) {
-    if (e instanceof TypeError) {
+    if (e instanceof TypeError || e instanceof Deno.errors.PermissionDenied) {
```
Comment (dsherret) [inline ext/node/polyfills/_process/process.ts:42]: I don't think we want to catch these and return undefined to fix this issue.

**Comment by dsherret** [comment]
Oh, I just realized that this one still shows the permission prompt. This seems better than https://github.com/denoland/deno/pull/20213

**Review by dsherret — APPROVED** [review]
LGTM! I think this is a better approach than the other PR, which would cause errors in existing code.

**Comment by kt3k** [comment]
@iuioiua Nice work!

**Pull request merged**